ddl/db_test.go

```diff
@@ -6662,3 +6662,20 @@ func (s *testSerialSuite) TestTruncateAllPartitions(c *C) {
 	tk1.MustExec("alter table partition_table truncate partition all;")
 	tk1.MustQuery("select count(*) from partition_table;").Check(testkit.Rows("0"))
 }
+
+func (s *testSerialSuite) TestIssue23872(c *C) {
```

Comment:
can you please add a test like `create table t(a int default 1, primary key(a));`